run/data/data.go

```diff
@@ -48,32 +48,72 @@ type Schema struct {
-	Type         string      // the converted name of the type
-	DBType       string      // the original type of the column in the DB
-	IsArray      bool        // true if the column type is an array
-	Length       int         // non-zero if the type has a length (e.g. varchar[16])
-	UserDefined  bool        // true if the type is user-defined
-	Nullable     bool        // true if the column is not NON NULL
-	HasDefault   bool        // true if the column has a default
-	IsPrimaryKey bool        // true if the column is a primary key
-	Orig         interface{} `yaml:"-" json:"-"` // the raw database column data
+	Table                               *Table                    `yaml:"-" json:"-"` // the table this column is in
+	Name                                string                    // the converted name of the column
+	DBName                              string                    // the original name of the column in the DB
+	Type                                string                    // the converted name of the type
+	DBType                              string                    // the original type of the column in the DB
+	IsArray                             bool                      // true if the column type is an array
+	Length                              int                       // non-zero if the type has a length (e.g. varchar[16])
+	UserDefined                         bool                      // true if the type is user-defined
+	Nullable                            bool                      // true if the column is not NON NULL
+	HasDefault                          bool                      // true if the column has a default
+	IsPrimaryKey                        bool                      // true if the column is a primary key
+	IsForeignKey                        bool                      // true if the column is a foreign key
+	IsForeignKeyReference               bool                      // true if the column is referenced by a foreign key
+	ForeignColumn                       *ForeignColumn            // foreign key database definition
+	ForeignKeyReferences                []string                  // all database names of foreign keys referencing this column
+	ForeignColumnsByForeignKeyReference map[string]*ForeignColumn // all columns referring to this column
+	Orig                                interface{}               `yaml:"-" json:"-"` // the raw database column data
+}
+
+type ForeignTable struct {
+	Name             string
+	TableName        string
+	ForeignTableName string
+	Table            *Table `yaml:"-" json:"-"`
+	ForeignTable     *Table `yaml:"-" json:"-"`
+}
+
+// Foreign Column contains the definition of a database foreign key
+type ForeignColumn struct {
+	Name                     string // the original name of the foreign key constraint in the db
+	ColumnName               string
+	ForeignColumnName        string
+	UniqueConstraintPosition int     // the position of the unique constraint in the db
```

Comment:
What's this used for?

Reply:
Each Foreign Key has a Unique Key on the other side. In the case of multi-column Foreign Keys, they have an order to them. The only use for them that I can tell is if you wanted to be able to template something that only used the first `x` number of keys to do a search. This would let know which keys you could use and still hit that index.

It's definitely not going to be a common use case. Up to you whether it stays or not.